haproxy_agent.py

```diff
@@ -504,6 +511,1078 @@ async def frp_auth(request: web.Request):
+        try:
+            inspect_url = f"http://{DOCKER_API_HOST}:{docker_engine_port}/containers/{container_name}/json"
+            async with session.get(inspect_url) as resp_inspect:
+                if resp_inspect.status == 200:
+                    inspect_data = await resp_inspect.json()
+                    original_state_running = inspect_data.get("State", {}).get("Running", False)
+                elif resp_inspect.status == 404:
+                    LOGGER.error("Container '%s' not found for certificate installation.", container_name)
+                    raise web.HTTPNotFound(text=f"Container '{container_name}' not found.")
+                else:
+                    error_text = await resp_inspect.text()
+                    LOGGER.error(
+                        "Failed to inspect container '%s' (status %s): %s",
+                        container_name,
+                        resp_inspect.status,
+                        error_text,
+                    )
+                    raise web.HTTPServiceUnavailable(text=f"Failed to inspect container: Status {resp_inspect.status}")
+
+            if not original_state_running:
+                LOGGER.info(
+                    "Container '%s' is not running. Attempting to start for certificate installation.", container_name
+                )
+                start_url = f"http://{DOCKER_API_HOST}:{docker_engine_port}/containers/{container_name}/start"
+                async with session.post(start_url) as resp_start:
+                    if resp_start.status not in (204, 304):
+                        error_text = await resp_start.text()
+                        LOGGER.error(
+                            "Failed to start container '%s' for cert install (status %s): %s",
+                            container_name,
+                            resp_start.status,
+                            error_text,
+                        )
+                        raise web.HTTPServiceUnavailable(text=f"Failed to start container: Status {resp_start.status}")
+                    LOGGER.info("Container '%s' started/is running for certificate installation.", container_name)
+
+            exit_code, os_info_content = await _execute_command_in_container_simplified(
+                session, docker_engine_port, container_name, ["cat", "/etc/os-release"]
+            )
+            if exit_code != 0:
+                LOGGER.error(
+                    "Failed to get OS info from container '%s'. Exit code: %s, Raw Output: %s",
+                    container_name,
+                    exit_code,
+                    os_info_content,
+                )
+                raise web.HTTPInternalServerError(
+                    text=f"Failed to get OS info. Exit: {exit_code}. Output: {os_info_content[:200]}"
+                )
+            LOGGER.info("OS Info for container '%s':\n%s", container_name, os_info_content.strip())
+
+            if payload.system_certs_bundle:
```

Comment:
wdyt of extracting `if payload.system_certs_bundle` and `if payload.install_frp_certs` into their own functions?
then early returns could be used like so the logic is a little easier to grasp and follow.
```py
if not certs_to_install:
    LOGGER.info(
        "No individual certificates parsed from system_certs_bundle for container '%s'.",
        container_name,
    )
    return

tar_bytes = _create_tar_archive_in_memory(certs_to_install)
...
```

Reply:
yeah yeah not adding an endpoint but just plain functions for system certs and frp certs so we have the return ability. It may simplify reading a little.